server/src/maps/mod.rs

```diff
@@ -0,0 +1,305 @@
+    for ((x_index, y_index), tile_img) in results {
+        image::imageops::overlay(
+            img,
+            &tile_img,
+            x_index * 512 - (x_img_koords as i64),
+            y_index * 512 - (y_img_koords as i64),
+        );
+    }
+}
+
+fn center_to_top_left_coordinates(x_pixels: u32, y_pixels: u32) -> (u32, u32) {
+    // the center coordniates are usefull for orienting ourselves in one tile,
+    // but for drawing them, top left is better
+    let y_to_img_border = 512 + y_pixels;
+    let y_img_koords = y_to_img_border - (630 - 125) / 2;
+    let x_to_img_border = 512 * 2 + x_pixels;
+    let x_img_koords = x_to_img_border - 1200 / 2;
+    (x_img_koords, y_img_koords)
+}
+
+fn is_in_range(x_pixels: u32, y_pixels: u32, x_index: u32, y_index: u32) -> bool {
+    let x_in_range = x_pixels <= (x_index + 1) * 512 && x_pixels + 1200 >= x_index * 512;
+    let y_in_range = y_pixels <= (y_index + 1) * 512 && y_pixels + (630 - 125) >= y_index * 512;
+    x_in_range && y_in_range
+}
+
+#[cfg(test)]
+mod range_tests {
+    use super::*;
+
+    fn assert_range_eq(
+        x_pixels: u32,
+        y_pixels: u32,
+        expected_x: (u32, u32),
+        expected_y: (u32, u32),
+    ) {
+        for x in 0..10 {
+            for y in 0..10 {
+                let (x_min, x_max) = expected_x;
+                let (y_min, y_max) = expected_y;
+                let expected_result = x <= x_max && x >= x_min && y <= y_max && y >= y_min;
+                assert_eq!(is_in_range(x_pixels, y_pixels, x, y), expected_result);
+            }
+        }
+    }
+
+    #[test]
+    fn ranged_test() {
+        assert_range_eq(0, 0, (0, 2), (0, 0));
+        assert_range_eq(0, 513, (0, 2), (1, 1));
+        assert_range_eq(512 / 2, 0, (0, 2), (0, 0));
+        assert_range_eq(512 / 2, 512 / 2, (0, 2), (0, 1));
+    }
+}
+
+fn lat_lon_to_xyz(lat_deg: f64, lon_deg: f64) -> (f64, f64, u32) {
+    let zoom = 16;
+    let lat_rad = lat_deg.to_radians();
+    let n = 2_u32.pow(zoom) as f64;
+    let xtile = (lon_deg + 180.0) / 360.0 * n;
+    let ytile = (1.0 - lat_rad.tan().asinh() / std::f64::consts::PI) / 2.0 * n;
+    (xtile, ytile, zoom)
+}
+
+lazy_static! {
+    static ref CANTARELL_BOLD: Font<'static> =
+        Font::try_from_bytes(include_bytes!("font/Cantarell-Bold.ttf")).unwrap();
+    static ref CANTARELL_REGULAR: Font<'static> =
+        Font::try_from_bytes(include_bytes!("font/Cantarell-Regular.ttf")).unwrap();
+}
+
+fn draw_bottom(data: &MapInfo, img: &mut image::RgbaImage) {
+    // draw background white
+    for x in 0..1200 {
+        for y in 630 - 125..630 {
+            img.put_pixel(x, y, Rgba([255, 255, 255, 255]));
+        }
+    }
+    // add our logo so the bottom
+    let logo = image::open("src/maps/logo.png").unwrap();
+    //let logo_height = logo_width / 200.832 * 32.115;
+    image::imageops::overlay(img, &logo, 10, 630 - (125 / 2) - (logo.height() as i64 / 2));
```

Comment:
```suggestion
    image::imageops::overlay(
        img,
        &logo,
        15,
        630 - (125 / 2) - (logo.height() as i64 / 2) + 9,
    );
```
I have slightly adjusted the position of the logo:
![image](https://user-images.githubusercontent.com/7429408/190113870-32b558ec-a61b-4d1c-98c1-3ab97e38caee.png)